File: dodian-backend/dodian-server/src/main/java/net/dodian/uber/game/model/item/GameItem.java
package net.dodian.uber.game.model.item;

import net.dodian.uber.game.Server;

public class GameItem {

    private final int id;
    private int amount;
    private final boolean stackable;

    public GameItem(int id, int amount) {
        stackable = Server.itemManager.isStackable(id);
        this.id = id;
        this.amount = amount;
    }

    public int getId() {
        return id;
    }

    public int getAmount() {
        return amount;
    }

    public void addAmount(int amount) {
        this.amount += amount;
    }

    public void removeAmount(int amount) {
        this.amount -= amount;
    }

    public boolean isStackable() {
        return stackable;
    }

}

File: game-server/src/main/kotlin/net/dodian/uber/game/model/item/GameItem.kt
package net.dodian.uber.game.model.item

import net.dodian.uber.game.Server

class GameItem(
    @JvmField val id: Int,
    @JvmField var amount: Int,
) {
    private val stackable: Boolean = Server.itemManager?.isStackable(id) ?: false

    fun getId(): Int = id

    fun getAmount(): Int = amount

    fun addAmount(amount: Int) {
        this.amount += amount
    }

    fun removeAmount(amount: Int) {
        this.amount -= amount
    }

    fun isStackable(): Boolean = stackable
}
